### PR TITLE
1500 move datafetcher to lib react4xp

### PR DESCRIPTION
**Pull request by ComLock**
Needs changes after https://github.com/enonic/npm-react-components/pull/77 is released in beta

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -64,11 +65,29 @@ if ( project.hasProperty( "dev" ) || project.hasProperty( "development" ) )
...
-jar.dependsOn( rollup )
+compileTestJava.dependsOn npmBuild
+jar.dependsOn npmBuild
+javadoc.dependsOn npmBuild
+
+// task rollup( type: NpmTask ) {
+//     mustRunAfter( processResources )
+//     args = ['run', 'rollup']
+// }
+// jar.dependsOn( rollup )
```
Comment (rymsha) [inline build.gradle:90]:
```suggestion
```

File: build.gradle [hunk 2]
```diff
@@ -88,7 +107,8 @@ else
     npmInstall.inputs.files( "package.json", "package-lock.json" )
     // npmInstall.outputs.dir("node_modules")
     npmInstall.outputs.file file( "package-lock.json" )
-    rollup.dependsOn( npmInstall )
+    // rollup.dependsOn( npmInstall )
```
Comment (rymsha) [inline build.gradle:110]:
```suggestion
```

**Review by alansemenov — CHANGES_REQUESTED** [review]
I see this was done in master a while ago, so this is not strictly related to this PR, but we need to clean this up before the release:

1. Jest is configured (jest.config, package.json, /jest/ folder etc) but doesn't seem to be used.
2. Bun is configured but doesn't seem to be used
3. You are using deno for tests but you don't install deno in package.json. We need to make sure tests are run on every build (`npmBuild` task depends on `test` task in build.gradle) as we do in other projects.

File: package.json [hunk 3]
```diff
@@ -25,44 +25,44 @@
...
+    "@enonic-types/global": "^7.15.0-A1",
+    "@enonic-types/lib-content": "^7.15.0-A1",
+    "@enonic-types/lib-io": "^7.15.0-A1",
+    "@enonic-types/lib-node": "^7.15.0-A1",
+    "@enonic-types/lib-portal": "^7.15.0-A1",
+    "@enonic-types/lib-schema": "^7.15.0-A1",
     "@enonic-types/lib-static": "^2.0.0",
     "@enonic/mock-xp": "^1.0.0",
+    "@enonic/react-components": "^6.0.0-B1",
     "@enonic/react4xp": "^5.0.4",
```
Comment (alansemenov) [inline package.json:37]: Shouldn't there be changes (6.0 version) in `@enonic/react-components` as well?

File: src/main/resources/types/Request.d.ts [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
-	}
-	host?: string
-	method?: string
-	mode?: 'edit'|'inline'|'live'|'preview'
-	params?: Params
-	path?: string
-	pathParams?: PathParams
-	rawPath?: string
-}
+export type {Request} from '@enonic-types/core';
```
Comment (rymsha) [inline src/main/resources/types/Request.d.ts:1]: One liner file

File: tsup.config.ts [hunk 5]
```diff
@@ -48,116 +40,120 @@ export default defineConfig((options) => {
...
-			format: [
-				// 'cjs',
-				// 'esm',
-				'iife' // tsup has no umd
-			],
-			minify,
-			outDir: 'build/resources/main/assets/react4xp',
-			platform: 'browser',
-			sourcemap: minify,
+	// if (options?.entry?.['client']) {
```
Comment (rymsha) [inline tsup.config.ts:59]: Dead code?

File: src/main/resources/types/Response.d.ts [hunk 6]
```diff
@@ -1,33 +1,36 @@
-import type {PageContributions} from './PageContributions';
-
+export type {
```
Comment (rymsha) [inline src/main/resources/types/Response.d.ts:1]: One line file

**Comment by ComLock** [comment]
> I see this was done in master a while ago, so this is not strictly related to this PR, but we need to clean this up before the release:
> 
> 1. Jest is configured (jest.config, package.json, /jest/ folder etc) but doesn't seem to be used.
or write more tests using it
> 2. Bun is configured but doesn't seem to be used
or write more tests using it
> 3. You are using deno for tests but you don't install deno in package.json. We need to make sure tests are run on every build (`npmBuild` task depends on `test` task in build.gradle) as we do in other projects.
Deno is not a node package. Deno is a node replacement.

File: package.json (hunk 3 above)
Comment (ComLock) [reply package.json:37]: ?

File: src/main/resources/types/Request.d.ts (hunk 4 above)
Comment (ComLock) [reply src/main/resources/types/Request.d.ts:1]: Removed

File: src/main/resources/types/Component.d.ts [hunk 7]
```diff
@@ -1,41 +1,34 @@
-import type {
-	Component as ComponentFromPortal,
-	Region
-} from "@enonic-types/lib-portal";
+// import type { Region } from "@enonic-types/lib-portal";
+export type { Component, PageComponent, PartComponent, TextComponent } from "@enonic-types/lib-portal";
 
 export type ComponentType = 'fragment'|'layout'|'page'|'part'|'text'; // image? region?
```
[the remark this answers is not in the document]
Comment (ComLock) [reply src/main/resources/types/Component.d.ts:1]: Fixed

File: tsup.config.ts (hunk 5 above)
Comment (ComLock) [reply tsup.config.ts]: removed

File: package.json (hunk 3 above)
Comment (ComLock) [reply package.json:37]: Do you mean @enonic/react4xp? Yeah it looks like we haven't release it with rspack 

https://www.npmjs.com/package/@enonic/react4xp?activeTab=versions 6 months old

File: src/main/resources/types/Response.d.ts (hunk 6 above)
Comment (ComLock) [reply src/main/resources/types/Response.d.ts]: Removed

File: package.json (hunk 3 above)
Comment (ComLock) [reply package.json:37]: lib-react4xp doesn't use the build system though, only some constants

Comment (ComLock) [reply package.json]: Upgraded to @enonic/react4xp 6.0.0-B1

**Review by rymsha — APPROVED** [review]

**Review by alansemenov — APPROVED** [review]

**Pull request merged**